label field
label line
label obstacle
label ball
label[] other_labels

